pkg/prometheus/agent/operator.go

```diff
@@ -1360,14 +1365,21 @@ func (c *Operator) checkAgentAuthorization(ctx context.Context) (bool, error) {
 				}
 				ssarResponse, err = c.kclient.AuthorizationV1().SelfSubjectAccessReviews().Create(ctx, ssar, metav1.CreateOptions{})
 				if err != nil {
-					return false, err
+					return nil, err
 				}
 				if !ssarResponse.Status.Allowed {
-					return false, nil
+					missingPermissions[resource] = append(missingPermissions[resource], verb)
 				}
 			}
+			if len(missingPermissions[resource]) == 0 {
+				delete(missingPermissions, resource)
+			}
```

Comment:
we can remove this if we followed the pattern above, `missingPermission[resource]` won't be created if the append above wasn't called.